cmd/kube-controller-manager/app/core.go

```diff
@@ -688,30 +691,64 @@ func startGarbageCollectorController(ctx context.Context, controllerContext Cont
 	for _, r := range controllerContext.ComponentConfig.GarbageCollectorController.GCIgnoredResources {
 		ignoredResources[schema.GroupResource{Group: r.Group, Resource: r.Resource}] = struct{}{}
 	}
+
+	// Run and Sync dependencyGraphBuilder if not already
+	if !controllerContext.GraphBuilder.IsRunning() {
+		go controllerContext.GraphBuilder.Run(ctx)
+		go controllerContext.GraphBuilder.Sync(ctx, discoveryClient, garbagecollector.SyncPeriod)
+	}
+
+	attemptToDelete, attemptToOrphan, absentOwnerCache, eventBroadcaster := controllerContext.GraphBuilder.GetGraphResources()
 	garbageCollector, err := garbagecollector.NewGarbageCollector(
 		ctx,
 		gcClientset,
 		metadataClient,
 		controllerContext.RESTMapper,
 		ignoredResources,
 		controllerContext.ObjectOrMetadataInformerFactory,
-		controllerContext.InformersStarted,
+		controllerContext.GraphBuilder,
+		attemptToDelete,
+		attemptToOrphan,
+		absentOwnerCache,
+		eventBroadcaster,
 	)
 	if err != nil {
-		return nil, true, fmt.Errorf("failed to start the generic garbage collector: %v", err)
+		return nil, false, fmt.Errorf("failed to start the garbage collector: %w", err)
```

Comment:
true -> false looks incorrect.  This was still enabled.

Also, why the text change?